src/averageAges.js

```diff
@@ -55,7 +74,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenWithMother = [];
+
+  people.filter(person => person.mother && (!onlyWithSon
```

Comment:
filter returns the filtered array so please rewrite the logic without pushing into `childrenWithMother` array and directly return the needed data